lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -1414,14 +1414,20 @@ void PrimTupleIndexOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
     if (i >= (int64_t)tupleConstruct.elements().size())
       return failure();
 
-    Value element = tupleConstruct.elements()[i];
     // TODO: We should have a clear picture of whether we want to consistently
     // allow refinement, and where. It seems desirable to require precise
     // type equality for TupleConstruct / TupleIndex, but that might break
     // things.
```

Comment:
nit: The "desirable to require precise type equality" part no longer applies

Reply:
I think it actually does still seem desirable, but is not the current state.